certificate/views.py

```diff
@@ -2618,3 +2619,141 @@ def create_python_workshop_certificate(certificate_path, name, qrcode, type, pap
+                if not present:
+                    short_key = serial_key[0:num]
+                    uniqueness = True
+                else:
+                    num += 1
+            qrcode = 'Verify at: http://fossee.in/certificates/verify/{0} '.format(short_key)
+            details = {'name': name,  'serial_key': short_key, 'email': email}
+            certificate = create_scipy_certificate_2017(certificate_path, details,
+                    qrcode, attendee_type, paper, workshop, file_name)
+            if not certificate[1]:
+                    certi_obj = Certificate(name=name, email=email, serial_no=serial_no,
+                            counter=1, workshop=workshop, paper=paper, serial_key=serial_key, short_key=short_key)
+                    certi_obj.save()
+                    return certificate[0] #render(request, 'scipy_download_2017.html')
+
+        if certificate[1]:
+            _clean_certificate_certificate(certificate_path, file_name)
+            context['error'] = True
+            return render_to_response('scipy_download_2017.html', context, ci)
+    context['message'] = ''
+    return render_to_response('scipy_download_2017.html', context, ci)
+
+
+@csrf_exempt
+def create_scipy_certificate_2017(certificate_path, name, qrcode, attendee_type, paper, workshop, file_name):
+    error = False
+    try:
+        template = 'template_SPC2017%scertificate' % attendee_type
+        download_file_name = 'SPC2017%scertificate.pdf' % attendee_type
+        template_file = open('{0}{1}'.format\
+                (certificate_path, template), 'r')
+        content = Template(template_file.read())
+        template_file.close()
+        if attendee_type == 'P' or attendee_type == 'T':
+            content_tex = content.safe_substitute(name=name['name'].title(),
+                    serial_key=name['serial_key'], qr_code=qrcode)
+        else:
+            content_tex = content.safe_substitute(name=name['name'].title(),
+                        serial_key=name['serial_key'], qr_code=qrcode, paper=paper)
+        create_tex = open('{0}{1}.tex'.format\
+                (certificate_path, file_name), 'w')
+        create_tex.write(content_tex)
+        create_tex.close()
+        return_value, err = _make_certificate_certificate(certificate_path, attendee_type, file_name)
+    
+
+        if return_value == 0:
+            pdf = open('{0}{1}.pdf'.format(certificate_path, file_name) , 'r')
+            response = HttpResponse(content_type='application/pdf')
+            response['Content-Disposition'] = 'attachment; \
+                    filename=%s' % (download_file_name)
+            response.write(pdf.read())
+            _clean_certificate_certificate(certificate_path, file_name)
+            return [response, False]
+        else:
+            error = True
+    except Exception, e:
+        error = True
+    return [None, error]
+    '''pdf = open('{0}{1}.pdf'.format(certificate_path, file_name) , 'r')
```

Comment:
remove this string if not used(Please do not push commented code).

Reply:
ok sir, Done